src/ssl/rustls.rs

```diff
@@ -23,20 +18,20 @@ fn read_test_cert(name: &str) -> Result<Vec<u8>> {
 
 fn root_store() -> Result<rustls::RootCertStore> {
     let mut root_store = rustls::RootCertStore::empty();
-    root_store.add_server_trust_anchors(webpki_roots::TLS_SERVER_ROOTS.0.iter().map(|ta| {
+    root_store.add_server_trust_anchors(webpki_roots::TLS_SERVER_ROOTS.iter().map(|ta| {
         rustls::OwnedTrustAnchor::from_subject_spki_name_constraints(
-            ta.subject,
-            ta.spki,
-            ta.name_constraints,
+            ta.subject.to_vec(),
+            ta.subject_public_key_info.to_vec(),
+            match ta.name_constraints.as_ref() {
+                Some(v) => Option::Some(v.to_vec()),
+                None => Option::None,
```

Comment:
I think this can be written shorted like `ta.name_constraints.as_ref().map(|c| c.to_vec())`